algorithms/src/std_algorithms/impl/Kokkos_RandomAccessIterator.hpp

```diff
@@ -63,6 +63,10 @@ class RandomAccessIterator<::Kokkos::View<DataType, Args...>> {
   using pointer           = typename view_type::pointer_type;
   using reference         = typename view_type::reference_type;
 
+#ifdef KOKKOS_ENABLE_SYCL
+  using is_passed_directly = std::true_type;
+#endif
```

Comment:
What is that for?

Reply:
As the pull request description says, this is needed to tell `oneDPL` not to assume that the data is on the host but on the device.

Reply:
Please add a comment in the code to capture that this tag is picked up by oneDPL with a reference to https://github.com/uxlfoundation/oneDPL/blob/a045eac689f9107f50ba7b42235e9e927118e483/include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h#L210-L214